### PR TITLE
pygobject3 3.30.0

### PR DESCRIPTION
**Pull request by tschoonj**

### CODE REVIEW

**Comment by tschoonj** [comment]
@DomT4 now that the build system of this formula has been moved to `meson`, I wonder what to do with the `python` dependency... It is required at build time in order to run `meson`, but it is also `:recommended` as it is possible to turn off the python3 bindings and build exclusively the python 2.7 bindings...

I put in `depends_on "python" => [:build, :recommended]`, but I doubt this is correct... Perhaps I should make the python3 bindings mandatory?

File: Formula/pygobject3.rb [hunk 1]
```diff
@@ -16,22 +15,33 @@ class Pygobject3 < Formula
   option "without-python", "Build without python3 support"
   option "with-python@2", "Build with python2 support"
 
+  depends_on "meson" => :build
+  depends_on "pkg-config" => :build
```
Comment (javian) [inline Formula/pygobject3.rb]: These dependencies should be listed in alphabetical order. `pkg-config` is also listed twice.

File: Formula/pygobject3.rb [hunk 2]
```diff
@@ -16,22 +15,32 @@ class Pygobject3 < Formula
...
   option "with-python@2", "Build with python2 support"
 
-  depends_on "pkg-config" => :build
-  depends_on "libffi" => :optional
   depends_on "glib"
+  depends_on "gobject-introspection"
+  depends_on "libffi" => :optional
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
```
Comment (DomT4) [inline Formula/pygobject3.rb]: Deps are structured by type & then alphabetically within those types, which I appreciate may be confusing. Essentially `:build` deps first, then mandatory dependencies, then recommended dependencies, then optional dependencies, then `if build.with?` dependencies.

File: Formula/pygobject3.rb [hunk 3]
```diff
@@ -16,22 +15,32 @@ class Pygobject3 < Formula
   option "without-python", "Build without python3 support"
   option "with-python@2", "Build with python2 support"
 
-  depends_on "pkg-config" => :build
-  depends_on "libffi" => :optional
   depends_on "glib"
+  depends_on "gobject-introspection"
+  depends_on "libffi" => :optional
```
Comment (DomT4) [inline Formula/pygobject3.rb]: Is this new or in the recursive tree somewhere already?

File: Formula/pygobject3.rb [hunk 4]
```diff
@@ -16,22 +15,32 @@ class Pygobject3 < Formula
...
-                            "PYTHON=#{python}"
-      system "make", "install"
-      system "make", "clean"
+    Language::Python.each_python(build) do |python, version|
+      mkdir "build#{version}" do
+        system "meson", "--prefix=#{prefix}",
+                        "-Dpycairo=true",
+                        "-Dpython=#{python}",
+                        ".."
+        # avoid linking against python framework
```
Comment (DomT4) [inline Formula/pygobject3.rb]: This should maybe be reported upstream, if this is the issue I think it is.

Comment (tschoonj) [reply Formula/pygobject3.rb]: Done at https://gitlab.gnome.org/GNOME/pygobject/issues/253

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-core/commit/ee44ad9b9c734acd1d491155a10e88d95ed345b6. Thanks!

**Pull request closed without merging**